src/main/kotlin/com/openlattice/shuttle/payload/XmlPayload.kt

```diff
@@ -0,0 +1,85 @@
+package com.openlattice.shuttle.payload
+
+import com.ctc.wstx.stax.WstxInputFactory
+import com.ctc.wstx.stax.WstxOutputFactory
+import com.fasterxml.jackson.dataformat.xml.XmlFactory
+import org.slf4j.LoggerFactory
+import java.io.File
+import java.io.InputStream
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.util.stream.Stream
+import javax.xml.stream.XMLStreamConstants.CHARACTERS
+import javax.xml.stream.XMLStreamConstants.END_ELEMENT
+
+data class XmlPayload(val directoryPath: String, val startTagName: String = DEFAULT_START_TAG) : Payload {
+
+    private val logger = LoggerFactory.getLogger(XmlPayload::class.java)
```

Comment:
For some reason, this adds 5.5s of time to the `testPayloadWithNoTag` unit test :(